File: win-prog/ProgWinCS/Essential Structures/FormSize/FormSize.cs
//---------------------------------------
// FormSize.cs © 2001 by Charles Petzold
//---------------------------------------
using System;
using System.Drawing;
using System.Windows.Forms;

class FormSize: Form
{
     public static void Main()
     {
          Application.Run(new FormSize());
     }
     public FormSize()
     {
          Text = "Form Size";
          BackColor = Color.White;
     }
     protected override void OnMove(EventArgs ea)
     {
          Invalidate();
     }
     protected override void OnResize(EventArgs ea)
     {
          Invalidate();
     }
     protected override void OnPaint(PaintEventArgs pea)
     {
          Graphics grfx = pea.Graphics;
          string   str  = "Location: "        + Location        + "\n"   +
                          "Size: "            + Size            + "\n"   +
                          "Bounds: "          + Bounds          + "\n"   +
                          "Width: "           + Width           + "\n"   +
                          "Height: "          + Height          + "\n"   +
                          "Left: "            + Left            + "\n"   +
                          "Top: "             + Top             + "\n"   +
                          "Right: "           + Right           + "\n"   +
                          "Bottom: "          + Bottom          + "\n\n" +
                          "DesktopLocation: " + DesktopLocation + "\n"   +
                          "DesktopBounds: "   + DesktopBounds   + "\n\n" +
                          "ClientSize: "      + ClientSize      + "\n"   +
                          "ClientRectangle: " + ClientRectangle;

          grfx.DrawString(str, Font, Brushes.Black, 0, 0);
    }
}

File: win-prog/ProgWinVB/Essential Structures/FormSize/FormSize.vb
'-----------------------------------------
' FormSize.vb (c) 2002 by Charles Petzold
'-----------------------------------------
Imports System
Imports System.Drawing
Imports System.Windows.Forms

Class FormSize
    Inherits Form

    Shared Sub Main()
        Application.Run(New FormSize())
    End Sub

    Sub New()
        Text = "Form Size"
        BackColor = Color.White
    End Sub

    Protected Overrides Sub OnMove(ByVal ea As EventArgs)
        Invalidate()
    End Sub

    Protected Overrides Sub OnResize(ByVal ea As EventArgs)
        Invalidate()
    End Sub

    Protected Overrides Sub OnPaint(ByVal pea As PaintEventArgs)
        Dim grfx As Graphics = pea.Graphics
        Dim str As String = _
            "Location: " & Location.ToString() & vbLf & _
            "Size: " & Size.ToString() & vbLf & _
            "Bounds: " & Bounds.ToString() & vbLf & _
            "Width: " & Width.ToString() & vbLf & _
            "Height: " & Height.ToString() & vbLf & _
            "Left: " & Left.ToString() & vbLf & _
            "Top: " & Top.ToString() & vbLf & _
            "Right: " & Right.ToString() & vbLf & _
            "Bottom: " & Bottom.ToString() & vbLf & vbLf & _
            "DesktopLocation: " & DesktopLocation.ToString() & vbLf & _
            "DesktopBounds: " & DesktopBounds.ToString() & vbLf & vbLf & _
            "ClientSize: " & ClientSize.ToString() & vbLf & _
            "ClientRectangle: " & ClientRectangle.ToString()

        grfx.DrawString(str, Font, Brushes.Black, 0, 0)
    End Sub
End Class
